### PR TITLE
Refactor and test readGenerator

### PR DESCRIPTION
**Pull request by dcolligan**
Issue #296

### CODE REVIEW

File: ga4gh/backend.py [hunk 1]
```diff
@@ -214,66 +214,16 @@ def variantSetsGenerator(self, request):
...
-                    equalPositionsToSkip += 1
-                else:
-                    equalPositionsToSkip = 0
-                nextPageToken = "{}:{}".format(
-                    getPosition(nextReadAlignment), equalPositionsToSkip)
-            yield readAlignment, nextPageToken
-            readAlignment = nextReadAlignment
+        readAlignment = self._positionIterator(
+            request, iterator, startPosition, equalPositionsToSkip)
+        for alignment, nextPageToken in self._yieldReads(
```
Comment (jeromekelleher) [inline ga4gh/backend.py]: I think we can just 
```python
return self._yieldReads(yadda, yadda)
```

**Comment by jeromekelleher** [comment]
This looks good @dcolligan, definitely a step in the right direction. I'm sorry I haven't had a chance to review it properly today, will try and get to it tomorrow.

**Comment by dcolligan** [comment]
That's fine, but there's not much to think about here.  It's just breaking up too-long methods into more manageable chunks and adding tests.

**Comment by dcolligan** [comment]
Closing PR to make a new one against `develop`

**Pull request closed without merging**